.pre-commit-config.yaml

```diff
@@ -0,0 +1,30 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
+    hooks:
+    -   id: trailing-whitespace
+
+-   repo: local
+    hooks:
+    # Run astyle over the staged files with c and h extension found in the directories
+    # listed in the files regex pattern. Ignoring the files in the exclude pattern.
+    - id: format-source
+      name: Formatting source files
+      entry: astyle --options=scripts/code-format-per-file.cfg
```

Comment:
Is there a reason that we can't just use `scripts/code-format.cfg`? It would be nice not to have multiple configs with the same information. I was able to unify the config file for the tests by adding `--ignore-exclude-errors` to the options and putting all the exclude cases together.

Reply:
Let me try adding the `--ignore-exclude-errors` to the entry command and I will report back, you're right about only having one configuration file.

Reply:
I'm getting this error

```console
Formatting source files..................................................Failed
- hook id: format-source
- duration: 0.02s
- exit code: 1

Recursive option with no wildcard
Did you intend quote the filename
Artistic Style has terminated
```

Reply:
We can move the `--recursive` option to `code-format.py` like this:
```c
os.system('astyle --options=code-format.cfg "../src/*.c,*.h" --recursive')
```

Reply:
See 5b336a9 and 2deb141, do they look ok?